### DISCUSSION TITLE
How to display search results in html with django

### DISCUSSION THREAD

**Discussion by micaelagimenez** [community/community#21484] [Other Feature Feedback, Questions, & Ideas]
<p>Hello, I am writing a search app that returns a movie from ghibli studio once you search for the title. The app it’s connected to the ghibli API and it works because whenever I type the title in the search bar, I get the result back but Only in my terminal. I need to display the results in the html but what I’ve tried so far has not worked.<br>
I’ve been told to do this in my html file:</p>
<pre><code class="lang-auto">{% if movie %}

{% for movie in movies %}

&lt;ul&gt;

&lt;li&gt;Title: {{ movie\_data.title }}&lt;/li&gt;

&lt;li&gt;Release date: {{ movie\_data.release\_date }}&lt;/li&gt;

&lt;li&gt;Director: {{ movie\_data.director }}&lt;/li&gt;

&lt;li&gt;Description: {{movie\_data.description}}&lt;/li&gt;

&lt;/ul&gt;

{% endfor %}

{% endif %}
</code></pre>
<p>This doesn’t work. This is my view:</p>
<pre><code class="lang-auto">from django.shortcuts import render, HttpResponse
from core.models import Movie
from django.views.generic import ListView
import requests

def index(request):
    movies = []
    
    if request.method == 'POST':
        
        film_url = 'https://ghibliapi.herokuapp.com/films/'
        
        search_params = {
            'films' : 'title',
            'films' : 'description',
            'films' : 'director',
            'films' : 'release_date',
            'q' : request.POST['search']
            
            }

        
        r = requests.get(film_url, params=search_params)
        results = r.json()
        print(results)

        for result in results:
             movie_data = {
                'Title' : result['title'],
                'Release_date': result['release_date'],
                'Director' : result['director'],
                'Producer' : result['producer'],
                'Description' : result['description']
            }

        movies.append(movie_data)  
        print(movie_data) 
     

    context = {
    'movies' : movies
    }
        
    return render(request,'core/index.html', context)
</code></pre>
<p>This is my html:</p>
<pre><code class="lang-auto">{% load static %}
&lt;!DOCTYPE html&gt;
&lt;html&gt;
&lt;head&gt;
    &lt;title&gt;Ghibli Studio | Movies&lt;/title&gt;
    &lt;link rel="stylesheet" href="{% static 'core/main.css' %}"&gt;
&lt;/head&gt;
    &lt;body&gt;
      &lt;div class=" header"&gt;
         
      &lt;/div&gt;
      &lt;div class="wrap"&gt;
          &lt;form action='/' method="POST"&gt;
              {% csrf_token %}
              &lt;div class="search"&gt;

                  &lt;input type="text" name="search" class="searchTerm" placeholder="Type movie name" &gt;
                  &lt;link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"&gt;
                  &lt;button type="submit" class="searchButton"&gt;

                      &lt;i class="fa fa-search" style="font-size:24px"&gt;&lt;/i&gt;
                  &lt;/button&gt;

              &lt;/div&gt;

          &lt;/form&gt;
          {% if movie %}
           {% for movie in movies %}
          &lt;ul&gt;
              &lt;li&gt;{{ movie }}&lt;/li&gt;
          &lt;/ul&gt;
           {% endfor %}
        
          {% endif %}

      &lt;/div&gt;
&lt;/body&gt;
&lt;/html&gt;

</code></pre>
<p>Any help would be very much appreciated!</p>

**Comment by airtower-luna** [comment]
<p>What’s the</p>
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="127195">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<p><code>{% if movie %}</code></p>
</blockquote>
</aside>
<p>block for? Your context doesn’t seem to contain a <code>movie</code> variable, so that might already be the reason that the enclosed <code>for</code> block isn’t rendering. It seems like</p>
<pre><code class="lang-auto">{% if movies %}
</code></pre>
<p>would make more sense. :wink:</p>

**Comment by micaelagimenez** [comment]
<p>Whoa, it was that easy! It worked! Thank you so much :smile:</p>

**Comment by micaelagimenez** [comment]
<p>Would you happen to know how I can display the result in the same way it appears in my movie_data dictionary?<br>
I’m trying this but it displays nothing if I do this, whereas it does display the whole block if I just do the {{movie}} thing, but it’s hard to understand since everything is together:</p>
<pre><code class="lang-auto">{% if movie %}
 {% for movie in movies %}
    &lt;ul&gt;
        &lt;li&gt;Director: {{ movie.director }}&lt;/li&gt;
        &lt;li&gt;Title: {{movie.title}}&lt;/li&gt;
        &lt;li&gt;Year: {{movie.release_date}}&lt;/li&gt;
        &lt;li&gt;Description: {{movie.description}}&lt;/li&gt;
    &lt;/ul&gt;
      {% endfor %}
    {% endif %}
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>That should work in principle, try capitalizing the same way as in the <code>movie_data</code> dict (e.g. <code>{{ movie.Title }}</code>).</p>

**Comment by micaelagimenez** [comment]
<p>Thank you so much again! That solved it!  :smile:</p>
